src/index.ts

```diff
@@ -59,10 +60,7 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
       if (fs.existsSync(userNpmrcPath)) {
         core.info("Found existing user .npmrc file");
         const userNpmrcContent = await fs.readFile(userNpmrcPath, "utf8");
-        const authLine = userNpmrcContent.split("\n").find((line) => {
-          // check based on https://github.com/npm/cli/blob/8f8f71e4dd5ee66b3b17888faad5a7bf6c657eed/test/lib/adduser.js#L103-L105
-          return /^\s*\/\/registry\.npmjs\.org\/:[_-]authToken=/i.test(line);
-        });
+        const authLine = extractAuthTokenLine(userNpmrcContent);
```

Comment:
You have mentioned that the previous version of the code goes in to the `else` branch and that creates a problem. I don't understand this. Even if it goes there... it will just append a different token to ur `.npmrc` file and that's about it. Your correct token should still be used when publishing to a custom artifactory